cmd/mdatagen/internal/sampleconnector/internal/metadata/generated_metrics_test.go

```diff
@@ -141,7 +167,7 @@ func TestMetricsBuilder(t *testing.T) {
 				assert.Equal(t, defaultMetricsCount, ms.Len())
 			}
 			if tt.metricsSet == testDataSetAll {
-				assert.Equal(t, allMetricsCount, ms.Len())
+				assert.Equal(t, (allMetricsCount/2)+1, ms.Len())
```

Comment:
This calculation isn't generic. We cannot assume that all receivers have exactly **1** metric with no attributes.

We should not change how treat allMetricsCount at all. We need to keep incrementing it per metric, not per datapoint. You already verify number of datapoint per metric below, which is good